src/widgets/modals/SearchForm.js

```diff
@@ -68,8 +71,24 @@ export const SearchForm = ({ dataSource }) => {
   }, [formConfig]);
 
   const renderItem = useMemo(
-    () => ({ item }) => <SearchListItem key={item.id} listItem={item} listConfig={listConfig} />,
-    [listConfig]
+    () => ({ item }) => {
+      const { id: key } = item;
+      const onPress = () => {
```

Comment:
Should separate this out from within this function so there isn't a new reference each time - so then the `SearchListItem` component can properly be memoized. With this layout there's no caching at all

Reply:
Agree 100%, this is a really bad cludge! Should be fixed in #2563.